src/app/components/admin/admin.component.html

```diff
@@ -1,3 +1,43 @@
-<div>
-    <p class="text text-lg-center text-success">Home</p>
-</div>
\ No newline at end of file
+<h1 class="text-center"> Admin Portal</h1>
+
+<div class="container">
+
+<div class="row">
+  <div class="offset-lg-1 offset-sm-1 col-lg-9 col-sm-10">
+   <input class="form-control" type="text" name="search" [(ngModel)]="searchText" autocomplete="off" placeholder="Search by Username">
+
+ </div>
+  <div class="col-lg-2 col-sm-1">
+     <button class="btn btn-primary" type="button" (click)="searchUser()">search</button>
+    </div>
+</div>
+
+<br>
+
+  <table class="table table-bordered">
+    <tr>
+      <th> ID </th>
+      <th> UserName </th>
+      <th> First Name </th>
+      <th> Last Name </th>
+      <th> Email </th>
+      <th> Phone Number </th>
+      <th> Batch Number</th>
+      <th> is Active </th>
+      <th> Ban </th>
+    </tr>
+
+    <tr *ngFor="let get of listofUsers">
+      <td>{{get.userId}}</td>
+      <td>{{get.userName}}</td>
+      <td>{{get.firstName}}</td>
+      <td>{{get.lastName}}</td>
+      <td>{{get.email}}</td>
+      <td>{{get.phoneNumber}}</td>
+      <td *ngIf="get.batch">{{get.batch.batchNumber}}</td>
```

Comment:
This ngIf makes things render inconsistently if there is no batch; puts "isActive" value in the batch column and the ban button in the "isActive" column

Reply:
we can use a ternary here to check if the batch exists and assigning the string interpolated value to the batch number if it does, or "n/a" if it doesnt